internal/websocket/ws.go

```diff
@@ -284,7 +289,21 @@ func (ws *Websocket) run(ctx context.Context) {
 			// Read loop
 			for {
 				var msg wrp.Message
+				ctx, cancel := context.WithTimeout(ctx, ws.inactivityTimeout)
```

Comment:
You can move the `cancel()` call here to ensure that any previous context got cleaned up prior to reading.  Just do something like:

```
if cancel != nil {
    cancel()
}
```

That's not perfect, as it means that if you exit the loop below, the context is still active.  but you can take care of that by having a parent context and doing a `defer parentCancel()` at the top of this function.